src/storage/db_interface_frontend.py

```diff
@@ -290,3 +292,13 @@ def find_missing_analyses(self):
             for entry in self.file_objects.find(query, {'_id': 1}):
                 missing_analyses.setdefault(firmware_uid, set()).add(entry['_id'])
         return missing_analyses
+
+    def find_failed_analyses(self) -> Dict[str, List[str]]:
+        query_result = self.file_objects.aggregate([
+            {'$project': {'analysis': {'$objectToArray': '$processed_analysis'}}},
+            {'$unwind': '$analysis'},
+            {'$match': {'analysis.v.failed': {'$exists': 'true'}}},
+            {'$group': {'_id': '$analysis.k', 'UIDs': {'$addToSet': '$_id'}}},
+        ], allowDiskUse=True)
+        plugin_to_uid_list = {entry['_id']: entry['UIDs'] for entry in query_result}
```

Comment:
`return {entry['_id']: entry['UIDs'] for entry in query_result}`

Reply:
added a doc string so that it is still clear that the result is grouped by plugin

Reply:
Accepted. Though you arguably could have just renamed entry to plugin ; )